### ISSUE TITLE
homebrew package dependencies versions log

### ISSUE THREAD

**Issue by shipmints** [jimeh/build-emacs-for-macos#113]
I've been using your Emacs 29.4 build. There have been bug fixes to the tree-sitter library since 29.4 was built. I can't find a log of what homebrew package versions for library dependencies (or even which specific gcc which can be critical due to compiler bugs) were used in a build.

I'd like to suggest that you add a build artifact to the pipeline with documentation for homebrew versions.

Further, I wonder if there's value in using a Brewfile with pinned versions for "production" builds and adding an extra build indicator; e.g. rather than just Emacs-29.4, they could be labeled as Emacs-29.4-jimeh-1 or -2 or -3 where each build has specific dependencies documented.

**Comment by jimeh** [comment]
Thanks for raising this. Yes I definitely want to do this, and ideally I’d like to switch to Nix as well. I had not realized caldwell is using Nix, so that might be a good reference.

I’ll see if I can get a list of exact relevant homebrew package versions generated as part of the build. I can’t promise a timeline at the moment, but I’m hoping I’ll enough spare time soon and get it sorted.

The next phase I’d like to focus on after that would be Nix :)

**Comment by shipmints** [comment]
At the very least, could add the build artifact that shows the homebrew versions of the downloaded dependencies. When weird things arise, it'll be a reference that saves us all time.

**Comment by jimeh** [comment]
This is somewhat addressed with the switch to Nix, as that locks all built-time deps to exact versions via the `flake.lock` file. And for the sake of visibility I have a Makefile target that also dumps all package names and versions into `flake.pkgs`.

Ultimately though, I want to have a look at GitHub's [Artifact Attestations](https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds) at some point and see if I can add it for builds produced by the workflows in [jimeh/emacs-builds](https://github.com/jimeh/emacs-builds).

**Comment by shipmints** [comment]
Indeed, nix > homebrew. I greatly dislike homebrew's total lack of reproducibility across systems and across time and coupled with homebrew's very early deprecation of platform support (Monterey not supported any longer despite it being the most widely used production macOS that I'm aware of where people have a choice), it makes hb a pain in the arse. If nix (better with guix, for sure) was a bit more accessible, I'd strongly consider it for home use. I rabbit hole I might have to dive into soon.
